src/api/content.ts

```diff
@@ -87,19 +88,89 @@ async function loadDocument(filePath: string) {
   }
 }
 
-export async function syncContent() {
-  const filePaths = await glob(`${CONTENT_DIR}/*.md`);
+export async function syncContent(contentDir = CONTENT_DIR) {
+  const filePaths = await glob(`${contentDir}/*.md`);
   filePaths.forEach(filePath => {
     taskQueue.pushTask(() => loadDocument(filePath));
   });
-  // TODO: handle deleted files
 }
 
-export function watchContent() {
-  const refreshContentDebounced = debounce(syncContent, 5000, {
-    trailing: true,
-  });
-  fs.watch(CONTENT_DIR, () => {
-    refreshContentDebounced();
-  });
+export function validateDocumentPayload(data: string) {
+  parseBody(data);
+  const fmResult = fm<object>(data);
+
+  const hasAttributes = Object.keys(fmResult.attributes).length;
+  if (!hasAttributes) {
+    throw new Error('invalid empty markdown head file.');
+  }
+
+  const hasBody = fmResult.body.replace(/\r?\n/ig, '').trim();
+  if (!hasBody) {
+    throw new Error('invalid empty markdown body file.');
+  }
+
+  return fmResult;
+}
+
+export function validateDocumentName(file: { originalname: string }) {
+  const hasMDExtension = path.extname(file.originalname) === '.md';
+
+  if (!hasMDExtension) {
+    throw new Error(`can't upload a none markdown file.`);
```

Comment:
This function can only meaningfully be called by the upload code as this error message makes that assumption. The function would be more reusable if it just returned a boolean (rather than throwing an error) indicating whether the file name was correct or not and let the caller decide how to handle the error.

Basically, the file name should consist of three parts, using the period `.` as a separator, and the last part should be `md`.  Therefore the simplest code I could think of would be:

```ts
export function validateDocumentName(file: { originalname: string }) {
  const parts = file.originalname.split('.');
  return parts.length !== 3 || parts[2] !== 'md';
}
```